File: go/pe002.go
// http://projecteuler.net/problem=2
package main

import "fmt"

func main() {
	sum := 0
	for x, y := 0, 1; x < 4e6; x, y = y, x+y {
		if x%2 == 0 {
			sum += x
		}
	}
	fmt.Println(sum)
}

File: swift/pe002.swift
// http://projecteuler.net/problem=2

var sum = 0
for var x = 0, y = 1; x < 4_000_000; (x, y) = (y, x + y) {
    if x % 2 == 0 {
        sum += x
    }
}
println(sum)
